python/src/robyn/visualization/plotter.py

```diff
@@ -0,0 +1,102 @@
+from robyn.modeling.feature_engineering import FeaturizedMMMData  # New import
+
+
+class FeaturePlotter:
+    """
+    A class for creating various plots related to feature engineering in the Robyn framework.
+    """
+
+    def __init__(self, mmm_data: MMMData, hyperparameters: Hyperparameters):
+        """
+        Initialize the FeaturePlotter class.
+
+        Args:
+            mmm_data (MMMData): The input data and specifications for the MMM.
+            hyperparameters (Hyperparameters): The hyperparameters for the model.
+        """
+        self.mmm_data = mmm_data
+        self.hyperparameters = hyperparameters
+
+    def plot_adstock(self, channel: str) -> plt.Figure:
+        """
+        Plot the adstock transformation for a specific channel.
+
+        Args:
+            channel (str): The name of the channel to plot adstock for.
+
+        Returns:
+            plt.Figure: A matplotlib Figure object containing the adstock plot.
+        """
+        pass
+
+    def plot_saturation(self, channel: str) -> plt.Figure:
+        """
+        Plot the saturation curve transformation for a specific channel.
+
+        Args:
+            channel (str): The name of the channel to plot saturation for.
+
+        Returns:
+            plt.Figure: A matplotlib Figure object containing the saturation curves plot.
+        """
+        pass
+
+    def plot_spend_exposure(self, featurized_data: FeaturizedMMMData, channel: str) -> plt.Figure:
+        """
+        Plot the relationship between spend and exposure for a given channel.
+
+        Args:
+            featurized_data (FeaturizedMMMData): The featurized data after feature engineering.
+            channel (str): The name of the channel being plotted.
+
+        Returns:
+            plt.Figure: A matplotlib Figure object containing the spend-exposure plot.
+        """
+        dt_mod = featurized_data.dt_mod
+        # Rest of the method implementation
+        pass
+
+    def plot_feature_importance(self, feature_importance: Dict[str, float]) -> plt.Figure:
+        """
+        Plot the importance of different features in the model.
+
+        Args:
+            feature_importance (Dict[str, float]): Dictionary of feature importances.
+
+        Returns:
+            plt.Figure: A matplotlib Figure object containing the feature importance plot.
+        """
+        pass
+
+    def plot_response_curves(self, featurized_data: FeaturizedMMMData) -> Dict[str, plt.Figure]:
+        """
+        Plot response curves for different channels.
+
+        Args:
+            featurized_data (FeaturizedMMMData): The featurized data after feature engineering.
+
+        Returns:
+            Dict[str, plt.Figure]: Dictionary mapping channel names to their respective response curve plots.
+        """
+        dt_mod = featurized_data.dt_mod
+        # Rest of the method implementation
+        pass
+
+    def save_plots(self, plots: Dict[str, plt.Figure], output_dir: str) -> None:
```

Comment:
can you remove this for now? create_all_plots that will create all plots and can take argument to export is better than this method, which takes plots that may or may not be created by this class and just exports it. But let's not even add it for now. robyn/utils/data_persistence_util.py will be single entry point for reading and writing to persistence layer.

Reply:
removing save_plots - to be used for data_persistence_util.py when we work on this interface.